textdb/textdb-common/src/main/java/edu/uci/ics/textdb/common/field/DataTuple.java

```diff
@@ -29,6 +29,8 @@ public IField getField(int index) {
 
     @Override
     public IField getField(String fieldName) {
+        //TODO the way we are storing the fields and schema, we need to linearly scan to get the field by name
+        //Do we need to change the Tuple design to support constant retrieval?

```

Comment:
@chenlica .. Currently our design of tuple is - we store list of attributes and list of fields separately and we rely on the attribute vs field mapping based on their order. So we need to linearly scan to find a field based on the field name..
To make it constant retrieval one solution would be to store the tuple data as Map<Attribute, IField>.  What do you think?